cache/disk/disk.go

```diff
@@ -69,17 +79,17 @@ func New(dir string, maxSizeBytes int64, proxy cache.CacheProxy) *DiskCache {
 	for _, c1 := range hexLetters {
 		for _, c2 := range hexLetters {
 			subDir := string(c1) + string(c2)
-			err := os.MkdirAll(filepath.Join(dir, cache.CAS.String(), subDir), os.ModePerm)
-			if err != nil {
-				log.Fatal(err)
-			}
-			err = os.MkdirAll(filepath.Join(dir, cache.AC.String(), subDir), os.ModePerm)
-			if err != nil {
-				log.Fatal(err)
-			}
-			err = os.MkdirAll(filepath.Join(dir, cache.RAW.String(), subDir), os.ModePerm)
-			if err != nil {
-				log.Fatal(err)
+
+			casDir := filepath.Join(dir, cache.CAS.String(), subDir)
```

Comment:
Should we also version the CAS dir? I think it's possible that we'll need to make changes to it eventually as well.

Reply:
I skipped versioning the CAS dir when I first implemented this because they are already stored in a natural representation on disk. But then I added support for different hash types, and forgot to revisit this choice.

Rethinking it now, CAS blobs should have a header to define the hash type, at the small cost of making it more difficult to verify blobs with the sha256sum command line utility. Maybe we should use the same header as for AC/RAW blobs, to simplify our code.

Reply:
A simple alternative could be to encode the hash function in the path string i.e. cas/sha256. Would that be too complex?

Reply:
Interesting idea- we could then just use a Digest proto as the header, and the blob offset is determined by the hash type in the filename.
 ````
ac.v2/sha256/<key>
ac.v2/md5/<key>
cas.c2/sha256/<key>
````

Reply:
It turns out that using a Digest like this is not so simple, since due to the SizeBytes field it doesn't have a fixed size on disk even if the Hash field has a constant length. So this would still require a prefix to store the length of the proto.

It would work well with a hand-rolled header though.